components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3001,24 +3001,32 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         const user = await this.checkAndBlockUser("deleteTeam");
         traceAPIParams(ctx, { teamId, userId: user.id });
 
-        await this.guardTeamOperation(teamId, "delete", "not_implemented");
+        const { team, members } = await this.guardTeamOperation(teamId, "delete", "delete");
 
-        const teamProjects = await this.projectsService.getTeamProjects(teamId);
-        teamProjects.forEach((project) => {
+        const projects = await this.projectsService.getTeamProjects(teamId);
```

Comment:
Right now, we don't have support for transactions across DB resources so we can't run a transaction across projects and teams